sdk/core/azure-core-perf/src/main/java/com/azure/core/perf/core/MockHttpClient.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.perf.core;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Function;
+
+public class MockHttpClient implements HttpClient {
```

Comment:
It'll be interesting to support our actual HTTP clients in the future as there are likely configuration we could apply to optimize them for the 80% case and offer canned configurations for other key scenarios. This may be closer to perf testing for the HTTP client implementations themselves.